src/client/errors.ts

```diff
@@ -3,79 +3,98 @@
  */
 
 /**
- * Error codes returned by actions involving creating applications.
+ * Error codes specific to the [[open]] function.
  */
 export enum OpenError {
     /**
-     * Indicates that an application cannot be found in the application directory.
+     * Indicates that an application of the provided name cannot be found in the application directory.
      */
     AppNotFound = 'AppNotFound',
+
     /**
-     * Indicates that an application cannot be started from an OpenFin manifest.
+     * In the case where the optional `context` argument is provided, indicates that the provided application was started, but threw an
+     * error when attempting to handle the provided context.
      */
-    ErrorOnLaunch = 'ErrorOnLaunch',
+    SendContextError = 'SendContextError',
+
     /**
-     * Indicates that an application was not created in a timely manner, and the provider timed out.
+     * In the case where the optional `context` argument is provided, indicates that the provided application was started, but a timeout
+     * was reached waiting for it to handle the provided context.
      */
-    AppTimeout = 'AppTimeout',
+    SendContextTimeout = 'SendContextTimeout',
+
     /**
-     * Currently unused.
+     * In the case where the optional `context` argument is provided, indicates that the provided application was started, but did not add
+     * a context handler.
      */
-    ResolverUnavailable = 'ResolverUnavailable',
+    SendContextNoHandler = 'SendContextNoHandler'
```

Comment:
This may be a question for the error code review story, but are these getting a bit too granular?

Is an application going to care about the distinction between these three codes? I definitely want us to be as detailed as possible in our error reporting, but the distinction between these three is something that could be conveyed in the message rather than by the code. The main purpose of the codes is to let applications handle errors programmatically, and I don't really see how an app would want/need to distinguish between these cases.

Feels like we're getting into "code and sub-code" territory here, which could be a decent compromise. Error code would be "problem starting app", sub-code would be "handler threw an error", "no handler registered", "handler timeout", etc.

Reply:
Will address in SERVICE-561